### PR TITLE
Cleanup and improvements to the Navigator

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 97716711

## Summary

This PR refactors the Navigator to make it more robust and cleans up the styling.

1. Spacings to be configurable via CSS variables.
2. Cleans up the sticky styling so its more robust
3. Allows specifying the filter positioning via a prop
4. Allow having none-truncated titles in nav items

## Dependencies

NA

## Testing

Steps:
1. Make sure there are no styling regressions.
1. Try to switch the filter positioning in the Vue dev tools by flipping the boolean prop on the `Navigator.vue`
2. Try to scroll fast and see if there are perf issues
3. Try in devtools to remove the `whitespace: nowrap` so titles overflow. Assert the navigator works as expected.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Comment by dobromir-hristov** [comment]
@swift-ci test

File: src/components/Navigator/NavigatorCard.vue [hunk 1]
```diff
@@ -82,37 +83,37 @@
...
+        <div class="filter-wrapper" v-if="!errorFetching">
+          <div class="navigator-filter">
+            <div class="input-wrapper">
+              <FilterInput
+                v-model="filter"
+                :tags="availableTags"
+                :selected-tags.sync="selectedTagsModelValue"
+                placeholder="Filter"
+                :should-keep-open-on-blur="false"
+                :position-reversed="!reverseFilterPosition"
```
Comment (marinaaisa) [inline src/components/Navigator/NavigatorCard.vue:95]: Why is `position-reversed` the opposite to `reverseFilterPosition`. Should it be the same value?
Maybe it's the naming what I find confusing

**Review by marinaaisa — DISMISSED** [review]
Left a question. Everything else looks good. I also tested it out

File: src/components/Navigator/NavigatorCard.vue (hunk 1 above)
Comment (dobromir-hristov) [reply src/components/Navigator/NavigatorCard.vue:95]: I needed a moment there 🤣 

In the `FilterInput` the default location of the dropdown is below, but for the Navigator, we render the filter down at the bottom, so we need the dropdown to render above.

But on mobile, the filter goes to the top of the `Navigator`, which means the `FilterInput` should render the dropdown in it's default position, which is down.

Comment (dobromir-hristov) [reply src/components/Navigator/NavigatorCard.vue:95]: Best case we rename `reverseFilterPosition` to `moveFilterToTop`, `renderFIlterAbove` or something else?

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Review by mportiz08 — APPROVED** [review]
Changes generally look ok to me and I'm not seeing any obvious issues testing this for regressions.

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Pull request merged**